### PR TITLE
ZCS-11116 : Update JDK to 17.0.2

### PR DESCRIPTION
**Pull request by umagmrit**

### CODE REVIEW

File: thirdparty/openjdk/Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ include $(PKG_ROOT)/../../package.def
 
 pvers := $(JDK_VERSION)
 pname := openjdk
-pfile := openjdk-13.0.1_linux-x64_bin.tar.gz
+pfile := openjdk-17.0.2_linux-x64_bin.tar.gz
```
Comment (silentsakky) [inline thirdparty/openjdk/Makefile:7]: should this use JDK_SUB_TREE / JDK_UPDATE /	JDK_BUILD / JDK_TAG / JDK_VERSION ?

File: versions.def [hunk 2]
```diff
@@ -71,11 +71,11 @@ HTTPD_VERSION		:= 2.4.53
...
 IMGK_VERSION		:= 6.7.7-2
 
-JDK_SUB_TREE		:= 8u
-JDK_UPDATE		:= 172
-JDK_BUILD		:= b01
-JDK_TAG			:= jdk8u$(JDK_UPDATE)-$(JDK_BUILD)
-JDK_VERSION		:= 1.8.0u$(JDK_UPDATE)$(JDK_BUILD)
+JDK_SUB_TREE		:= 17
+JDK_UPDATE		:= 17.0
+JDK_BUILD		:= 2
```
Comment (silentsakky) [inline versions.def:76]: any reason we are not using 17.0.3.1 ?

File: versions.def [hunk 3]
```diff
@@ -71,11 +71,11 @@ HTTPD_VERSION		:= 2.4.53
 
 IMGK_VERSION		:= 6.7.7-2
 
-JDK_SUB_TREE		:= 8u
-JDK_UPDATE		:= 172
-JDK_BUILD		:= b01
-JDK_TAG			:= jdk8u$(JDK_UPDATE)-$(JDK_BUILD)
-JDK_VERSION		:= 1.8.0u$(JDK_UPDATE)$(JDK_BUILD)
+JDK_SUB_TREE		:= 17
```
Comment (silentsakky) [inline versions.def:74]: also JDK 18 is out, any reason to not pick the latest one?
I think it's too late to ask this question now, so if we think we can still go to JDK 18 then let's file separate ticket to upgrade that in future

Comment (ghen2) [reply versions.def:74]: JDK 17 is an LTS release, supported until 2029, whereas 18 is a regular release, only supported until 19 comes out.

File: versions.def (hunk 2 above)
Comment (umagmrit) [reply versions.def:76]: when we pickup this task, 17.0.3.1 not yet released.

**Comment by stondino00** [comment]
Please upgrade to 17.0.3 the latest version.  Fixes numerous CVE's.

image

**Review by silentsakky — APPROVED** [review]

**Comment by silentsakky** [comment]
@stondino00 Thanks for the heads up, as we have already started with 17.0.2 version testing and we are close to releasing the changes, it would be hard to bump version at this time, but we will make sure to upgrade to latest 17.x release in the next iteration

**Comment by stondino00** [comment]
Please upgrade to 17.0.5.

image

image

**Comment by stondino00** [comment]
Oracle has again released updated java versions for Q1 2023 security fixes.  Please look at upgrading to latest.

**Comment by stondino00** [comment]
Just a reminder to upgrade to the latest java version.  Many security fixes over the last few quarters.

**Comment by stondino00** [comment]
Please upgrade java 17 in Zimbra 9.  Current version is over a year old. Below is the security release notes from April.


image

**Comment by stondino00** [comment]
Another quarter for Oracle security updates has been released.  This version is VERY old that's bundled with Zimbra 9 and 10.  PLEASE upgrade this and the mariadb bundled versions to fix MANY MANY vulnerabilities and bugs.

image

**Pull request merged**